### PR TITLE
Clarify how Skaffold resolves files

### PR DESCRIPTION
**Pull request by briandealwis**
**Related**: #5355

**Description**
Clarify where the `skaffold.yaml` is expected to be found and how referenced files and directories are resolved against the current working directory, except for build artifacts.

### CODE REVIEW

File: docs/content/en/docs/design/config.md [hunk 1]
```diff
@@ -25,6 +21,43 @@ read the configuration file from the current directory.
...
+is resolved relative to the `frontend` directory (i.e., `frontend/Dockerfile`),
+whereas the the Helm chart's location and its values-files are
+relative to the current directory in `helm/project`.
+```yaml
+apiVersion: skaffold/v2beta11
+kind: Config
+build:
+  artifacts:
+  - image: app
+    context: frontend
```
Comment (gsquared94) [inline docs/content/en/docs/design/config.md:47]: should we mention that if `context` is not explicitly defined then it's set to the CWD anyway?

File: docs/content/en/docs/design/config.md [hunk 2]
```diff
@@ -25,6 +21,43 @@ read the configuration file from the current directory.
...
+relative to the current directory _and not to the location of
+the Skaffold configuration file_.  There is one important exception:
+files referenced from a build artifact definition are resolved
+relative to the build artifact's _context_ directory.
+
+In the following config file, the `Dockerfile` for building `app`
+is resolved relative to the `frontend` directory (i.e., `frontend/Dockerfile`),
+whereas the the Helm chart's location and its values-files are
+relative to the current directory in `helm/project`.
+```yaml
```
Comment (gsquared94) [inline docs/content/en/docs/design/config.md:41]: also show the directory tree?
```
.
├── frontend
│   └── Dockerfile
├── helm
│   └── project
│       └── dev-values.yaml
└── skaffold.yaml
```

File: docs/content/en/docs/design/config.md [hunk 3]
```diff
@@ -25,6 +21,43 @@ read the configuration file from the current directory.
 
 You can [learn more]({{< relref "/docs/references/yaml" >}}) about the syntax of `skaffold.yaml`.
 
+Skaffold normally expects to find the configuration file as
+`skaffold.yaml` in the current directory, but the location can be
+overridden with the `--filename` flag.
+
+### File resolution
```
Comment (gsquared94) [inline docs/content/en/docs/design/config.md:28]: Can you add a `File resolution` section down below or mention here itself about configurations resolved as dependencies. The paths in those files are resolved relative to the directory containing the imported file.  So it's _recommended_ to have the `skaffold.yaml` file live in the root of the project and not in a nested directory.

**Review by briandealwis — COMMENTED** [review]
Oh jeepers, I missed your comments @gsquared94.  PTAL.

**Review by gsquared94 — APPROVED** [review]
lgtm!

**Comment by briandealwis** [comment]
Failure was due to Docker Hub`toomanyrequests`

**Pull request merged**